stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/externimpl/consumer/ArtemisMessageHandler.java

```diff
@@ -0,0 +1,276 @@
+import org.ballerinalang.messaging.artemis.ArtemisConstants;
+import org.ballerinalang.messaging.artemis.ArtemisUtils;
+import org.ballerinalang.model.types.BArrayType;
+import org.ballerinalang.model.types.BMapType;
+import org.ballerinalang.model.types.BStructureType;
+import org.ballerinalang.model.types.BType;
+import org.ballerinalang.model.types.BTypes;
+import org.ballerinalang.model.types.TypeTags;
+import org.ballerinalang.model.util.JSONUtils;
+import org.ballerinalang.model.util.JsonParser;
+import org.ballerinalang.model.util.XMLNodeType;
+import org.ballerinalang.model.util.XMLUtils;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BByte;
+import org.ballerinalang.model.values.BFloat;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.model.values.BValueArray;
+import org.ballerinalang.model.values.BXML;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Handler for Artemis messages.
+ *
+ * @since 0.995
+ */
+class ArtemisMessageHandler implements MessageHandler {
+    private static final Logger logger = LoggerFactory.getLogger(ArtemisMessageHandler.class);
+
+    private Resource onMessageResource;
+    private BMap<String, BValue> sessionObj;
+    private boolean autoAck;
+
+    ArtemisMessageHandler(Resource onMessageResource, BMap<String, BValue> sessionObj, boolean autoAck) {
+        this.onMessageResource = onMessageResource;
+        this.sessionObj = sessionObj;
+        this.autoAck = autoAck;
+    }
+
+    @Override
+    public void onMessage(ClientMessage clientMessage) {
+        List<ParamDetail> paramDetails = onMessageResource.getParamDetails();
+        int paramSize = paramDetails.size();
+        if (paramSize > 1) {
+            dispatchResourceWithDataBinding(clientMessage, paramDetails);
+        } else {
+            dispatchResource(clientMessage, createAndGetMessageObj(onMessageResource, clientMessage, sessionObj));
+        }
+
+    }
+
+    private void dispatchResourceWithDataBinding(ClientMessage clientMessage, List<ParamDetail> paramDetails) {
+        BValue[] bValues = new BValue[paramDetails.size()];
+        try {
+            bValues[1] = getContentForType(clientMessage, paramDetails.get(1).getVarType());
+            bValues[0] = createAndGetMessageObj(onMessageResource, clientMessage, sessionObj);
+            dispatchResource(clientMessage, bValues);
+        } catch (BallerinaException ex) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("The message received do not match the resource signature", ex);
+            }
```

Comment:
Shall we make the logger level to error? IMO we need to pass the error to the user.